# 例外处理

#### 知识点：

- try/catch/finally



#### 代码示例：

```kotlin
import java.lang.Exception
import java.lang.NumberFormatException

fun main() {
    val value = "10"
    var out: Int? = null

    // 异常处理
    try {
        out = Integer.parseInt(value)
    } catch (e: NumberFormatException) {
        println(">>>NumberFormatException<<<")
        println(e.message)
    } catch (e: Exception) {
        println(">>>Exception<<<")
        println(e.message)
    } finally {
        println("out is $out")
    }

    // 赋值处理
    var output = try {
        Integer.parseInt(value)
    } catch (e: Exception){
        -1
    }

    println("output is $output")
}
```

